base_classes/nyaml/NXopt_element.yaml

```diff
@@ -0,0 +1,38 @@
+category: base 
+doc: |
+  Properties of generic optical element in an experimental setup for optical spectroscopy.
+
+  
+NXopt_element:
+  previous_optical_elements: 

```

Comment:
If you write it without a type, it assumed that  `previous_optical_elements` is a string.

Reply:
this should be fine, as for each beam only one specific element should exist and this reference is done by a path to the respective element located at the NXinstrument level.